napari/_qt/layer_controls/qt_points_controls.py

```diff
@@ -283,29 +297,32 @@ def change_text_visibility(self, state):
         state : QCheckBox
             Checkbox indicating if text is visible.
         """
-        # needs cast to bool for Qt6
-        self.layer.text.visible = bool(state)
+        with self.layer.text.events.visible.blocker(
```

Comment:
Are you adding these because they're needed to avoid infinite recursion? Or if they're not needed right now, to prevent in the future? Or is this more about efficiency - i.e. blocking a callback we know doesn't need to be executed?

I removed this one (and its control equivalent) and didn't get infinite recursion.

My initial reaction is just to omit blockers if they're not needed, but I think the other two reasons are fair. Would still be good to clarify in this PR though.

Reply:
To prevent in the future; also, maybe we don't get *infinite* recursion cause we check somehwere (in the layer model, likely) if the value is the same, but we still get at least a double call to the setter, and potentially two events. There are several cases in the past where I was debugging events and found duplicated events. I think preventing these is valuable in case we (or a user) connects expensive callbacks. There's probably a cost to context managers, but hopefully that's negligible?